presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -689,9 +704,12 @@ protected Node visitSimpleGroupBy(SimpleGroupBy node, Void context)
 
         private boolean validateExpressionForGroupBy(Set<Expression> groupByOfMaterializedView, Expression expression)
         {
+            boolean willRewrite = expression instanceof FunctionCall
+                    && MaterializedViewUtils.rewriteNonAssociativeFunction((FunctionCall) expression, materializedViewInfo.getBaseToViewColumnMap()).isPresent();
```

Comment:
since we are adding this, should be check if the function is from supported list(including associative functions)